lib/search/presenters/spell_check_presenter.rb

```diff
@@ -3,13 +3,27 @@ module Search
     def present
       return [] unless any_suggestions?
 
-      es_response["suggest"]["spelling_suggestions"].first["options"].map { |option| option["text"] }
+      suggestions.map do |option|
+        if highlighted_suggestions?
+          { text: option["text"], highlighted: option["highlighted"] }
+        else
+          option["text"]
+        end
+      end
     end
 
   private
 
     def any_suggestions?
       es_response["suggest"] && es_response["suggest"].any?
     end
+
+    def highlighted_suggestions?
+      suggestions.detect { |option| option.has_key? "highlighted" }
+    end
+
+    def suggestions
+      es_response["suggest"]["spelling_suggestions"].first["options"]
```

Comment:
Is it possible `spelling_suggestions` won't exist or could be empty? If so I guess this would raise an error, but the logic here hasn't changed so it looks like it is fine?

Reply:
It would raise an error if evaluated when there are no suggestions, but `return [] unless any_suggestions?` happens before `suggestions` is used.